### PR TITLE
Upgrade to sundrio 0.30.0

### LINKED ISSUE #2924: Creating a CustomResourceInfo from a class doesn't result in the expected information
Properties don't seem to be available and scope is wrong, in particular.

### PR DESCRIPTION
**Pull request by iocanel**
## Description
Fixes #2924 

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by metacosm** [comment]
This change is needed for the CRD generator to work properly from the API instead of only from APT.

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/v1/decorator/EnsureSingleStorageVersionDecorator.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public EnsureSingleStorageVersionDecorator(String name) {
 
 	@Override
 	public void andThenVisit(CustomResourceDefinitionSpecFluent<?> spec, ObjectMeta resourceMeta) {
-    Predicate<CustomResourceDefinitionVersionBuilder> hasStorageVersion = CustomResourceDefinitionVersionFluentImpl::isStorage;
+    Predicate<CustomResourceDefinitionVersionBuilder> hasStorageVersion = CustomResourceDefinitionVersionFluentImpl::getStorage;
```
Comment (manusa) [inline crd-generator/api/src/main/java/io/fabric8/crd/generator/v1/decorator/EnsureSingleStorageVersionDecorator.java:37]: I'm worried about this, have the boolean getters changed from isXXX to getXXX after the Sundr.io upgrade?

Comment (metacosm) [reply crd-generator/api/src/main/java/io/fabric8/crd/generator/v1/decorator/EnsureSingleStorageVersionDecorator.java:37]: I think that was a semantic change that @iocanel did, unrelated to the upgrade

**Comment by manusa** [comment]
I'm comparing the generated-sources before and after the upgrade (just for the kubernetes-model-core module), and found a lot of differences.

I'm not sure if these are expected. It's affecting builder methods to set string values. Seems that before we could do stuff like this:
```java
withNewMyStringField(1);
withNewMyStringField('c');
// ...
```
and we no longer can.

image

**Comment by manusa** [comment]
Sorry, I got it the other way around :facepalm: . We've added new builder methods :+1:

**Review by manusa — APPROVED** [review]

**Comment by metacosm** [comment]
Closed in favor of #2971

**Pull request closed without merging**